sig-and-wg/sig-core/proposals/modularization.md

```diff
@@ -0,0 +1,72 @@
+# Kyma modularization - install components as add-ons
+
+Created on 2018-08-16 by Piotr Kopczynski (@PK85) and Piotr Bochyński (@pbochynski).
+
+## Status
+
+Proposed on 2018-08-16.
+
+## Motivation
+
+Kyma provides components as a one big pack to enable developers building something on cloud native stack. They work together and they expose the unified developer and business experience. Now, we need to step back and enable users to decide what they want to use Kyma for.
+
+The landscape around Kyma has changed. Kubernetes managed clusters get popular. Each big cloud infrastructure provider has own managed Kubernetes service (Google GKE, Amazon EKS, Microsoft AKS). This offering is evolving and customers can use more services included in the managed Kubernetes (like Istio - Service Mesh, Service Catalog, Logging). Developers work on bringing more features to transform Kubernetes from container scheduling system into complete platform solution. Kyma is not the only solution in this area. Knative is the new open source project started by Google that also aims to provide components to deliver cloud native software on top of Kubernetes. There is a strategic decision to run Kyma on top of Knative. All this development in the Kubernetes landscape shows the need of having flexible, loosely-coupled components to enable customers using best available solutions from infrastructure providers, open source projects and commercial solutions. In principle, Kyma offers a complete solution (batteries included), but it doesn't mean Kyma can't benefit from running on top of managed cluster, by integration with the already provided services.
+
+## Goal
```

Comment:
Additional goal is to: Improve speed of testing components and integrating them in Kyma. We need to be able to install kyma with only components that are modified and run only acceptance tests related to them.

Reply:
It just matters for local installation, right?

Reply:
for now we can say it matters for local as it would be mainly used for development on local, or for PRs within docker-in-docker. But if the modularization is effective and GKE and other providers fast enough, maybe we can stop maintaining docker-in-docker and do proper testing on a cluster for a PR - just an idea how we can profit in future